### ISSUE TITLE
Attention mask是不需要吗？

### ISSUE THREAD

**Issue by BaberMuyu** [CyberZHG/keras-bert#171]
你好，我在对比了kears-bert和google的bert实现后发现有一点不同，keares-bert中没有对padding的数据做attention mask，而google的bert中有这一部分实现。我在实验中也意外发现在短文本上使用bert时，不同的max_length（都会大于文本的最大长度）会对结果产生影响，而按照我原本的理解是不应该出现这种区别的。
![image](https://user-images.githubusercontent.com/21008660/69306356-21b41600-0c62-11ea-92b8-0f511b49ae28.png)

**Comment by CyberZHG** [comment]
https://www.tensorflow.org/guide/keras/masking_and_padding#masking

**Comment by BaberMuyu** [comment]
> https://www.tensorflow.org/guide/keras/masking_and_padding#masking
input_embedding = token_embedding +segment_embedding + position_embedding
token_embedding的padding部分被mask，但是segment_embedding 和position_emgedding的padding部分没有被mask
下面是我看到的链接资料（在代码实现部分有提到相关内容）和在bert源码中看到的部分代码
[https://kexue.fm/archives/4765](url)

`  if attention_mask is not None:
    # `attention_mask` = [B, 1, F, T]
    attention_mask = tf.expand_dims(attention_mask, axis=[1])

    # Since attention_mask is 1.0 for positions we want to attend and 0.0 for
    # masked positions, this operation will create a tensor which is 0.0 for
    # positions we want to attend and -10000.0 for masked positions.
    adder = (1.0 - tf.cast(attention_mask, tf.float32)) * -10000.0

    # Since we are adding it to the raw scores before the softmax, this is
    # effectively the same as removing these entirely.
    attention_scores += adder`

**Comment by CyberZHG** [comment]
https://github.com/keras-team/keras/blob/master/keras/layers/merge.py#L197

**Comment by BaberMuyu** [comment]
> https://github.com/keras-team/keras/blob/master/keras/layers/merge.py#L197

非常感谢

**Comment by 1535966643** [comment]
你好，上面的连接已经失效了，我怎么给keras-bert输入一个input_mask呢

**Comment by 1535966643** [comment]
您好，连接失效了，是不是keras-bert不需要再去做mask了呢？
